R/missing_path.R

```diff
@@ -0,0 +1,26 @@
+find_closest_path <- function(s, max_dist = 2, max_matches = 3) {
+  all_files <- list.files(path = getwd(), pattern = NULL, all.files = FALSE,
+                          full.names = TRUE, recursive = FALSE,
+                          ignore.case = FALSE, include.dirs = FALSE, no.. = FALSE)
+  dists <- stringdist::stringdist(s, all_files)
+  sorted_dists <- sort(dists, index.return=TRUE)
+  idxs_within_max <- sorted_dists$ix[sorted_dists$x <= max_dist]
+  if (length(idxs_within_max) == 0) { return(list()) }
+
+  files_within_max <- all_files[idxs_within_max]
+  files_within_max[1:min(length(files_within_max), max_matches)]
+}
+
+handle_no_path <- function(trace) {
+  missing_path <- trace[["trace"]][["calls"]][[1]][[2]]
```

Comment:
I think we also need to think through this behavior a little more. For example, consider if I do:

```r
load_file <- function() { read.csv("missing_file") }
```

Then if I do:

```r
trace <- rlang::last_trace()
missing_path <- trace$trace$calls[[1]][[2]]
```

This fails with the error `subscript out of bounds`. Because now the file is at the second layer of the trace, not the first.

At the very least, we should add logic that checks whether or not the indices `[[1]][[2]]` exist in the trace, and `return(FALSE)` otherwise.